src/test/java/org/kohsuke/github/GHBranchProtectionTest.java

```diff
@@ -73,4 +80,11 @@ public void testSignedCommits() throws Exception {
         protection.disableSignedCommits();
         assertFalse(protection.getRequiredSignatures());
     }
+
+    @Test
+    public void testGetProtection() throws Exception {
+        GHBranchProtection protection = branch.enableProtection().enable();
+        GHBranchProtection protectionTest = repo.getBranch(BRANCH).getProtection();
+        assertTrue(protectionTest instanceof GHBranchProtection);
```

Comment:
Even though the above branch master is already protected it still throws a message that the branch is not protected. It seems a bit strange
```   
 "body": "{\"message\":\"Branch not protected\",\"documentation_url\":\"https://developer.github.com/v3/repos/branches/#remove-branch-protection\"}",
```

Reply:
GitHub doesn't think it is protected.   The data files definitely show that.

Reply:
Is there any solution ? Because manually changing doesn't seem to work

Reply:
@sladyn98  
Not sure.  I'll take a look.